executor/write.go

```diff
@@ -169,6 +169,18 @@ func updateRecord(ctx context.Context, sctx sessionctx.Context, h kv.Handle, old
 		txnCtx := sctx.GetSessionVars().TxnCtx
 		if txnCtx.IsPessimistic {
 			txnCtx.AddUnchangedRowKey(unchangedRowKey)
+			// Update DeltaForTable
+			colSize := make(map[int64]int64, len(newData))
+			for id, col := range t.Cols() {
+				_, err := codec.EstimateValueSize(sc, newData[id])
+				if err != nil {
+					continue
+				}
+				// Nothing changed so size should be 0
+				colSize[col.ID] = 0
+			}
+			udpp := sctx.GetSessionVars().UseDynamicPartitionPrune()
+			txnCtx.UpdateDeltaForTable(t.Meta().ID, physicalID, 0, 0, colSize, udpp)
```

Comment:
@wjhuang2016 do I change `colSize` to nil good for here?

Reply:
You can use `map[int64]int64{}`.